javascript/ql/src/Statements/UseOfReturnlessFunction.ql

```diff
@@ -149,6 +149,57 @@ predicate voidArrayCallback(DataFlow::CallNode call, Function func) {
   )
 }
 
+
+/**
+ * Provides classes for working with various Deferred implementations. 
+ * It is a heuristic. The heuristic assume that a class is a promise defintion 
+ * if the class is called "Deferred" and the method `resolve` is called on an instance.
+ *  
+ * Removes some false positives in the js/use-of-returnless-function query.  
+ */
+module Deferred {
+  /**
+   * An instance of a `Deferred` class. 
+   * E.g. the result from `new Deferred()` or `new $.Deferred()`.  
```

Comment:
We don't generally use abbreviations like "e.g." in comments; see https://wiki.semmle.com/pages/viewpage.action?pageId=17105939 (internal link).